genie-web/src/main/java/com/netflix/genie/web/configs/GenieServicesAutoConfiguration.java

```diff
@@ -343,4 +346,41 @@ public AgentRoutingService agentRoutingService(
             genieHostInfo
         );
     }
+
+    /**
+     * Get an implementation of {@link JobCompletionService} if one hasn't already been defined.
+     * @param jobPersistenceService    The job persistence service to use
+     * @param jobSearchService         The job search service to use
+     * @param genieFileTransferService The file transfer service to use
+     * @param genieWorkingDir          Working directory for genie where it creates jobs directories.
+     * @param mailService              The mail service
+     * @param registry                 Registry
+     * @param jobsProperties           The jobs properties to use
+     * @param retryTemplate            The retry template
+     * @return an instance of {@link JobCompletionService}
+     * @throws GenieException if the bean fails during construction
+     */
+    @Bean
+    @ConditionalOnMissingBean(JobCompletionService.class)
```

Comment:
I kind of avoided doing this move in my refactoring because my hope was that most of the stuff from the V3 job path would be removed once we fully got v4 going. No big deal if you do this either way just telling you why I didn't bother.